sleap/io/dataset.py

```diff
@@ -795,13 +798,26 @@ def extract(self, inds, copy: bool = False) -> "Labels":
                 - `Labels.suggestions`
                 - `Labels.provenance`
         """
+        # Get subset of labeled frames.
         lfs = self.__getitem__(inds)
-        new_labels = type(self)(
+
+        # Get subset of videos.
+        videos_used = set([lf.video for lf in lfs])
+        videos = list(videos_used)
+
+        # Get subset of suggestions.
+        suggestions = [
+            suggestion
+            for suggestion in self.suggestions
+            if suggestion.video in videos_used
+        ]
```

Comment:
I added this pruning functionality here, but maybe we would rather have it in the ExportLabelsSubset command. Do we ALWAYS want to prune the videos and suggestions list to the subset of videos that the found `LabeledFrame`s uses?